adam/core/transforms.py

```diff
@@ -0,0 +1,126 @@
+import abc
+from typing import TypeVar
+
+import numpy as np
+
+T = TypeVar("T")
+
+
+class AbstractTranforms(abc.ABC):
+    def __init__(self) -> None:
+        pass
+
+    def R_from_axisAngle(self, axis, q):
```

Comment:
Since there are no private attributes, all these methods could become class methods (i.e. replace `self` with `cls` in body).

```suggestion
    @classmethod
    def R_from_axisAngle(cls, axis, q):
```